crates/ra_hir_expand/src/lib.rs

```diff
@@ -252,3 +268,14 @@ impl<T> Source<T> {
         db.parse_or_expand(self.file_id).expect("source created from invalid file")
     }
 }
+
+pub struct MacroDefIdWithAst {
```

Comment:
Note that without salsa db, we have to use a `ast::MacroCall` directly to build the macro rules. Thats why I add this structure.